pyproject.toml

```diff
@@ -43,6 +43,7 @@ numba = "^0.58.1"
 
 
 [tool.poetry.group.test.dependencies]
+hypothesis = "^6.92.2"
```

Comment:
I think adding hypothesis is the right idea here, however I don't have much experience with it. It might take me a while to thoroughly understand this

Reply:
No problem. Here are some links that I found helpful depending how far down the rabbit hole you want to go with it.
- [What is Property Based Testing? - Hypothesis](https://hypothesis.works/articles/what-is-property-based-testing/)
- [Quick start guide — Hypothesis 6.92.2 documentation](https://hypothesis.readthedocs.io/en/latest/quickstart.html)
- [Beyond Units by Hillel Wayne](https://youtu.be/MYucYon2-lk?si=wWcvg175QJr642i4)
- [Choosing properties for property-based testing | F# for fun and profit (fsharpforfunandprofit.com)](https://fsharpforfunandprofit.com/posts/property-based-testing-2/)

Reply:
Thanks! I've read the Hypothesis documentation and done some basic property based tests, but not much beyond that.